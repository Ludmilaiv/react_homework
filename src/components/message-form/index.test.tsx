import { MessageForm } from './';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('MessageForm', () => {
  it('render component', () => {
    const mockFunction = jest.fn();
    render(<MessageForm addMessage={mockFunction} />);
  });

  it('render with snapshot', () => {
    const mockFunction = jest.fn();
    const { asFragment } = render(<MessageForm addMessage={mockFunction} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render multiply components', () => {
    const mockFunction = jest.fn();
    render(
      <>
        <MessageForm addMessage={mockFunction} />
        <MessageForm addMessage={mockFunction} />
      </>
    );

    expect(screen.queryAllByRole('form').length).toBe(2);
  });
});
