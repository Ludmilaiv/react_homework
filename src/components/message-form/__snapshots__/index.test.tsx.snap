// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm render with snapshot 1`] = `
<DocumentFragment>
  <form
    action=""
    class="app__message message-function"
    role="form"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="outlined-basic"
        id="outlined-basic-label"
      >
        Автор
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="outlined-basic"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1in441m"
          >
            <span>
              Автор
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="outlined-basic"
        id="outlined-basic-label"
      >
        Сообщение
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="outlined-basic"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1ftyaf0"
          >
            <span>
              Сообщение
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      Отправить
    </button>
  </form>
</DocumentFragment>
`;
